cmd/kube-scheduler/app/server.go

```diff
@@ -169,11 +168,12 @@ func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, sched *
 	logger.Info("Golang settings", "GOGC", os.Getenv("GOGC"), "GOMAXPROCS", os.Getenv("GOMAXPROCS"), "GOTRACEBACK", os.Getenv("GOTRACEBACK"))
 
 	// Configz registration.
-	if cz, err := configz.New("componentconfig"); err == nil {
-		cz.Set(cc.ComponentConfig)
-	} else {
+	cz, err := configz.New("componentconfig")
```

Comment:
Hey, is there any logical change in the code that could make it easier to understand?

Reply:
I removed the 'else' block because the 'if' statement already returns an error if the condition is met. 
By eliminating the 'else' block, it simplifies the code structure and avoids unnecessary nesting, which can improve readability.

Reply:
hey, but this doesn't seem to simplify the logic

Reply:
Yeah, I would lay this out differently than before or after:

```
if cz, err := configz.New("componentconfig"); err != nil {
    // return...
} else {
    // use cz
}
```

Our brains are wired to handle for `err != nil` - anything else is a hazard

Reply:
Now I understand, thanks for the feedback.

Reply:
Thank you for the feedback! I’ve refactored the code as you suggested to limit the scope of `cz` and `err` and to improve readability with the `if err != nil` pattern. I appreciate the guidance and the suggestions on Go best practices. Please let me know if there’s anything else to adjust.